### DISCUSSION TITLE
about <Link /> usage

### DISCUSSION THREAD

**Discussion by TpTTT-T** [vercel/next.js#37034] [Help]
I had seen this suggestion some where, but I do not know why.

use this
```
<Link>
     <a>123</a>
</Link>
```

not this

```
<Link>
     123
</Link>
```

**Comment by icyJoseph** [comment]
I am not sure :) - but I think it had something to do with cloning the element inside, and support for various use cases. 

> - Styled-jsx was already mentioned above and it's also not possible to give the Link-Component a class.

That being said, there has been some incremental changes that make it so that you can just pass a string, for all cases, but also try to keep the current (legacy now I guess) behaviour. I fond out that about 5 years ago (time flies), the automatic wrapping of string children, and other things was added.

Take a look here: https://github.com/vercel/next.js/pull/36436

My take, is that, on this one, we should wait until there's **official communication** from the Next.js team.

## Edit

A colleague found this docs from v9, https://nextjs.org/docs/tag/v9.5.2/api-reference/next/link#using-a-component-that-supports-onclick:

> Link supports any component that supports the onClick event, in the case you don't provide an <a> tag, consider the following example:
>
>```jsx
> <Link href="/about">
>    <img src="/static/image.png" alt="image" />
> </Link>
> ```
> The child of Link is <img> instead of <a>. Link will send the onClick property to <img> but won't pass the href property.

So string children, and those that support the onClick event, native tags, will do the trick.

**Comment by GabenGar** [comment]
You have to do this if you want to pass `className` prop to the `<a>` element.
